contracts/contracts/Coordinator.sol

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity ^0.8.0;
+
+/**
+* @title CoordinatorV1
+* @notice Coordination layer for DKG-TDec
+*/
+contract CoordinatorV1 {
+
+    uint8 public constant DKG_SIZE = 8;
+
+    event NewRitual(uint32 indexed ritualID, address[] nodes);
+    event TranscriptPosted(uint32 indexed ritualID, address indexed node, bytes32 transcriptDigest);
+    event ConfirmationPosted(uint32 indexed ritualID, address indexed node, address[] confirmedNodes);
+
+    enum RitualState {
+        NON_INITIATED,
+        WAITING_FOR_TRANSCRIPTS,
+        WAITING_FOR_CONFIRMATIONS,
+        ENDED
+    }
+
+    // TODO: Find better name
+    struct Performance {
+        address node;
+        uint96 confirmedBy;
+        bytes32 transcript;
+    }
+
+    struct Ritual {
+        uint32 id;
+        uint32 initTimestamp;
+        Performance[DKG_SIZE] performance;
+    }
+
+    uint32 public immutable transcriptsWindow;
+    uint32 public immutable confirmationsWindow;
+    Ritual[] public rituals;
+
+    constructor(uint32 _transcriptsWindow, uint32 _confirmationsWindow){
+        transcriptsWindow = _transcriptsWindow;
+        confirmationsWindow = _confirmationsWindow;
+    }
+
+    function numberOfRituals() external view returns(uint256){
+        return rituals.length;
+    }
+
+    function initiateRitual(address[] calldata nodes) external {
+        // TODO: Check for payment
+        // TODO: Check for expiration time
+        // TODO: Improve DKG size choices
+        require(nodes.length == DKG_SIZE, "Invalid number of nodes");
+
+
+        uint32 id = uint32(rituals.length);
+        Ritual storage ritual = rituals.push();
```

Comment:
My solidity knowledge is lacking - if this function fails later on (eg. if the nodes aren't sorted properly), is this initialized ritual still stored, or not? If not still stored, great - nothing to see here.

Reply:
Contract calls are atomic in the EVM, so if the call fails due to a revert for example, then there's no state changes.